### PR TITLE
Interaction - Show all possible modes for an attachment

### PR DESCRIPTION
**Pull request by PabstMirror**
image

### CODE REVIEW

**Comment by Dystopian** [comment]
Tested, works. I would suggest to move these actions from `Detach` submenu to between `Attach` and `Detach` actions. It's much easier to select actions this way at least when menu is in list mode.

**Comment by mjc4wilton** [comment]
> Tested, works. I would suggest to move these actions from `Detach` submenu to between `Attach` and `Detach` actions. It's much easier to select actions this way at least when menu is in list mode.

Maybe make it so it goes Weapon Attachments -> [Array of attachments] -> [Mode, Detach, etc.]?

File: addons/interaction/functions/fnc_getWeaponAttachmentsActions.sqf [hunk 1]
```diff
@@ -49,8 +49,30 @@ params ["_unit"];
     // "detach" actions
     {
         if (_x isEqualTo "") then {continue};
+        private _originalAttachment = _x;
 
-        private _config = _cfgWeapons >> _x;
+        private _switchableAttachments = [_originalAttachment] call CBA_fnc_switchableAttachments;
+        private _convertToActions = [];
+        {
+            if (_x == _originalAttachment) then {continue};
```
Comment (Dystopian) [inline addons/interaction/functions/fnc_getWeaponAttachmentsActions.sqf:57]:
```suggestion
            if (_x == _originalAttachment) then {continue};
            if (getText(_CBA_PIPItems >> _x) == _originalAttachment) then {continue};
```
CBA PIP optics should be ignored too. CUP has many scopes like this (e.g. ACOG and SUSAT).

File: addons/interaction/functions/fnc_getWeaponAttachmentsActions.sqf [hunk 2]
```diff
@@ -49,8 +49,30 @@ params ["_unit"];
     // "detach" actions
     {
```
Comment (Dystopian) [inline addons/interaction/functions/fnc_getWeaponAttachmentsActions.sqf:50]:
```suggestion
    private _CBA_PIPItems = configFile >> "CBA_PIPItems";
    {
```

**Review by Dystopian — CHANGES_REQUESTED** [review]

**Comment by johnb432** [comment]
> Maybe make it so it goes Weapon Attachments -> [Array of attachments] -> [Mode, Detach, etc.]?

I fully agree with that, given we're adding some new actions. I went ahead and pushed the necessary changes.

Do we want to allow switching to PIP modes for scopes? To me it doesn't seem to be the worst idea, but I don't really use those scopes.

**Review by johnb432 — APPROVED** [review]
LGTM, but given I've rewritten part of the code, I'd like another person to take a second look at this.

**Comment by PabstMirror** [comment]
image

IMHO this might helpreadability to make the detach/attach sub-actions simpler
so they stand out from the switch action?

instead of `SPIR` -> `Detach: SPIR`
now just  `SPIR` -> `Detach`

**Comment by johnb432** [comment]
> IMHO this might helpreadability to make the detach/attach sub-actions simpler so they stand out from the switch action?
> 
> instead of `SPIR` -> `Detach: SPIR` now just `SPIR` -> `Detach`

Done.

**Pull request merged**